snowflake.go

```diff
@@ -121,9 +121,8 @@ func NewNode(node int64) (*Node, error) {
 		return nil, errors.New("Node number must be between 0 and " + strconv.FormatInt(n.nodeMax, 10))
 	}
 
-	var curTime = time.Now()
 	// add time.Duration to curTime to make sure we use the monotonic clock if available
-	n.epoch = curTime.Add(time.Unix(Epoch/1000, (Epoch%1000)*1000000).Sub(curTime))
```

Comment:
I believe the reason for how this works now is creating the value and adding to it forces Go to use a monotonic clock value.  Though, it seems like surely there's a better way.  We'd want to make sure any change also uses monotonic clock value.  I'll try to read the docs on this and see if I can confirm how it works.  If you (or anyone else) understands that part of Go better feel free to let me know! :)

Reply:
I spent some time reading up on Go time functions, and time.Unix() does not include the monotonic runtime value, so we wouldn't want to use it since further in the code the time.Since() calls needs that value to make sure our ID's are generated based off a monotonic time.

Reply:
Thanks, I have no idea about the monotonic clock, now I understand, and you are right.